### PR TITLE
java: Add async-profiler build logic, and upgrade it.

### PR DESCRIPTION
**Pull request by Jongy**
## Description
Like https://github.com/Granulate/gprofiler/pull/69 and https://github.com/Granulate/gprofiler/pull/68, for async-profiler.

## Motivation and Context
See mentioned PRs.

## How Has This Been Tested?
Our tests.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] I have updated the relevant documentation.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by guybortnikov — CHANGES_REQUESTED** [review]
Looks good,
delete the async-profiler section in `build.sh`

**Comment by Jongy** [comment]
> Looks good,
> delete the async-profiler section in `build.sh`

Hmm I was certain I had this committed. Good catch

**Review by guybortnikov — APPROVED** [review]

**Comment by Jongy** [comment]
Our previous release was built on `phusion/holy-build-box-64` which is based on CentOS 6 (though I don't know which libc its compiler uses). This PR changes async-profiler to use CentOS 7. I'll make sure that the resulting `libasyncProfiler.so` still loads successfully in old systems as it used to (e.g on CentOS 6). If needed, I'll downgrade the build to use CentOS 6.

**Comment by Jongy** [comment]
> Our previous release was built on `phusion/holy-build-box-64` which is based on CentOS 6 (though I don't know which libc its compiler uses). This PR changes async-profiler to use CentOS 7. I'll make sure that the resulting `libasyncProfiler.so` still loads successfully in old systems as it used to (e.g on CentOS 6). If needed, I'll downgrade the build to use CentOS 6.

When testing on CentOS 6, Java profiling failed because we couldn't execute `jattach` which needed a glibc 2.14 symbol. Technically `jattach` could be executed with `via_staticx=True` and it should solve the problem. But this means that async-profiler was built with an older libc altogether, so I prefer to keep the build old, as well, to avoid similar issues with `libasyncProfiler.so`.

**Review by guybortnikov — APPROVED** [review]

**Comment by Jongy** [comment]
:astonished: it passed. dumb CI. I have 1 small improvement for it, and we will merge after current release. I also want to test it on a real CentOS 6 machine, to ensure `libasyncProfiler.so` is loaded successfully.

**Comment by Jongy** [comment]
> I also want to test it on a real CentOS 6 machine, to ensure `libasyncProfiler.so` is loaded successfully.

Done. Works okay on CentOS 6 with:
```
centos@ip-172-31-28-21:~$ ldd --version
ldd (GNU libc) 2.12
Copyright (C) 2010 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```

**Comment by Jongy** [comment]
Ahhhh

```
chunk = {'error': 'failed to register layer: ApplyLayer exit status 1 stdout:  stderr: write /usr/local/openjdk-8/man/ja_JP.UT...er exit status 1 stdout:  stderr: write /usr/local/openjdk-8/man/ja_JP.UTF-8/man1/keytool.1: no space left on device'}}
```

File: tests/conftest.py [hunk 1]
```diff
@@ -112,9 +113,21 @@ def docker_client() -> DockerClient:
...
+@fixture(scope="session")
+def gprofiler_docker_image_resources(gprofiler_docker_image: Image):
+    """
+    Copy all resources from the built image.
+    """
+    copy_file_from_image(
+        gprofiler_docker_image,
+        os.path.join("/app", "gprofiler", "resources"),
+        os.path.join(os.path.dirname(gprofiler.__file__)),
+    )
```
Comment (guybortnikov) [inline tests/conftest.py:130]: delete the files after the tests cycle finish

Comment (Jongy) [reply tests/conftest.py:130]: nah, why

**Review by guybortnikov — APPROVED** [review]

**Pull request merged**